projects/lpc40xx_freertos/l5_application/gps/line_buffer.c

```diff
@@ -0,0 +1,69 @@
+/**
+ * @file line_buffer.c
+ *
+ * @brief This file maintains the APIs for managing line buffer to handle the received
+ *      string from GPS line by line.
+ */
+
+#include "line_buffer.h"
+
+void line_buffer__init(line_buffer_s *buffer, void *memory, size_t size) {
+  buffer->memory = memory;
+  buffer->max_size = size;
```

Comment:
`memset(buffer->memory, 0, buffer->max_size)`
you should reset the buffer memory in init